doc/tutorials/basic/build_todo.md

```diff
@@ -0,0 +1,137 @@
+# Build a Todo App
+
+In this section, we will work on building a *Todo App* together so that our wind turbine technicians can keep track of their tasks. As a team, we will collaborate to create an app that provides the following functionality:
+
+- Adding, removing, and clearing all tasks
+- Marking a task as solved
+- Keeping track of the number of completed tasks
+- Disabling or hiding buttons when necessary
+
+:::{note}
+When we ask everyone to *run the code* in the sections below, you may either execute the code directly in the Panel docs via the green *run* button, in a cell in a notebook, or in a file `app.py` that is served with `panel serve app.py --autoreload`.
+:::
+
+## Create the App
+
+```{pyodide}
+import panel as pn
+
+pn.extension(sizing_mode="stretch_width", design="material")
+
+BUTTON_WIDTH = 125
+
+# We use intslider to avoid teaching users pn.rx. Is that a good thing?
+state_changed_count = pn.widgets.IntInput()
+tasks = pn.Column()
+
+def update_state_changed_count(*args):
+    state_changed_count.value += 1
+
+def remove_task(task, *args):
+    index = tasks.index(task)
+    tasks.pop(index)
+
+def remove_all_tasks(*args):
+    tasks.clear()
+
+def create_task(text):
+    state = pn.widgets.Checkbox(align="center", sizing_mode="fixed")
+    content = pn.pane.Markdown(text)
+    remove = pn.widgets.Button(width=BUTTON_WIDTH, icon="trash", sizing_mode="fixed")
+    task = pn.Row(state, content, remove, sizing_mode="stretch_width")
+
+    pn.bind(remove_task, task, remove, watch=True)
+    # We have to bind the below after the above!
+    pn.bind(update_state_changed_count, state, remove, watch=True)
+
+    return task
+
+def add_task(text, *args):
+    if not text:
+        return
+
+    new_task = create_task(text)
+    tasks.append(new_task)
+
+    return tasks
+
+def get_state(*args):
+    total_tasks = len(tasks)
+    completed_tasks = sum(check[0].value for check in tasks)
+    return f"{completed_tasks} of {total_tasks} tasks completed"
+
+def can_add(value_input):
+    return not bool(value_input)
+
+def has_tasks(*args):
+    return len(tasks) > 0
+
+
```

Comment:
adding comment sections would hlep I think

```
# Add tasks

...
# Initialize Widgets
```
etc

Reply:
Agree. Its just not implemented yet in the `## Break it down` section just below.

There is a separate discussion [here](https://github.com/holoviz/panel/pull/6213#discussion_r1477064735) on whether the tutorials should lead up to a solution, or present the solution first and then break it down. It would be nice get agreement on that first before I finalize `build_todo.md`.